django/db/models/functions/window.py

```diff
@@ -0,0 +1,86 @@
+"""
+Window aggregate functions. They are implemented in such a way that it would be
+possible to easily use them in a expression of the type
+
+    WITHIN GROUP
+
+Not all of these functions are available in all implementations of SQL:2003. A
+flag in the feature-list for a database backend indicates availability.
+"""
+
+from django.db.models import Aggregate
+from django.db.models.fields import Field, FloatField, IntegerField
+
+__all__ = [
+    'RowNumber', 'Rank', 'DenseRank', 'Ntile', 'FirstValue',
+    'PercentRank', 'NthValue', 'LastValue', 'CumeDist', 'Lag',
+    'Lead'
+]
+
+
+class CumeDist(Aggregate):
+    function = 'CUME_DIST'
+    name = 'CumeDist'
+    _output_field = FloatField()  # or DecimalField()
+
+
+class DenseRank(Aggregate):
+    function = 'DENSE_RANK'
+    name = 'DenseRank'
+    _output_field = IntegerField()
+
+
+class FirstValue(Aggregate):
+    function = 'FIRST_VALUE'
+    _output_field = Field()
+
+
+class Lag(Aggregate):
+    function = 'LAG'
+    _output_field = Field()
+
+
+class LastValue(Aggregate):
+    function = 'LAST_VALUE'
+    _output_field = Field()
+
+
+class Lead(Aggregate):
+    function = 'LEAD'
+    name = 'Lead'
+    _output_field = Field()
+
+
+class NthValue(Aggregate):
+    function = 'NTH_VALUE'
+    name = 'NthValue'
+    _output_field = Field()
+
+
+class Ntile(Aggregate):
+    function = 'NTILE'
+    template = 'Ntile(%(num_buckets)d)'
+    _output_field = FloatField()  # or DecimalField
+
+    def __init__(self, num_buckets=1, **extra):
+        self.num_buckets = num_buckets
+        extra.update({'num_buckets': self.num_buckets})
+        super(Ntile, self).__init__(**extra)
+
+
+class Rank(Aggregate):
+    function = 'RANK'
+    name = 'Rank'
+    _output_field = IntegerField()
+
+
+class RowNumber(Aggregate):
+    function = 'ROW_NUMBER'
+    name = 'RowNumber'
+    _output_field = IntegerField()
+
+
+class PercentRank(Aggregate):
+    function = 'PERCENT_RANK'
+    name = 'PercentRank'
+    _output_field = FloatField()  # or DecimalField
```

Comment:
I'm not sure what the suggested change is. However, I noticed that functions are importable from `django.db.models.functions` which I've just done (may need some additional work).

The functions can be used without an expression, and with an expression, for instance in `WITHIN GROUP`.